mozdownload/scraper.py

```diff
@@ -819,11 +819,16 @@ def binary_regex(self):
 
     def build_filename(self, binary):
         """Return the proposed filename with extension for the binary."""
-        return '%(TIMESTAMP)s%(BRANCH)s%(DEBUG)s-%(NAME)s' % {
-            'TIMESTAMP': self.timestamp + '-' if self.timestamp else '',
+        if not self.timestamp and self.date:
+            date = self.date.strftime('%Y-%m-%d-')
+        else:
+            date = ''
+        # str(...) necessary to create str type else unicode --> test_tinderbox fail
+        return str('%(TIMESTAMP)s%(BRANCH)s%(DEBUG)s-%(NAME)s' % {
```

Comment:
I don't see why we have to use `str()` here. I would suggest that you get rid of it again, and check where the unicode characters are coming from. I believe its from the `self.date.strftime`call which you have added here.